Makefile.make

```diff
@@ -251,7 +252,7 @@ docclean:
 	rm -rf doc/sphinx/_build
 
 archclean: clean-ide optclean voclean plugin-tutorialclean
-	rm -rf _build
+	rm -rf _build _build_boot
```

Comment:
Is clean really supposed to remove dune's stuff?

Reply:
I don't know, `_build` was there so I added `_build_boot` too; cannot hurt.

Actually this line is coming from the times there were experimental support for `ocamlbuild`  (2012)